src/salte-auth.js

```diff
@@ -1,1226 +1,358 @@
- * @property {string} redirectUri         Optional; defaults to current window location.  Redirection URI to which the response will be sent. This URI MUST exactly match one of the Redirection URI values for the Client pre-registered at the OpenID Provider.
- * @property {string} loginResource       Optional; defaults to clientId.
- * @property {object} securedEndpoints    Optional.  Object containing one or more named endpoints (e.g. { "Endpoint": "ResourceId" }).  This list is used to identify endpoints to which tokens should be attached.  If "Endpoint" exists anywhere within the endpoint being called then it is considered to be a match.  If the endpoint being called doesn't match then either no token will be passed or, if the endpoint's hostname matches the application hostname, then the id_token will be passed.
- * @property {Array} anonymousEndpoints   Optional.  Array containing one or more endpoints to which a token need not be attached.  If the endpoint specified exists anywhere within the endpoint being called then it is considered to be a match.
- * @property {boolean} isAngular          Optional.  Used to tell this library whether it should handle the callback from the authorization server itself or bubble the response up to the main window's URL and let the Angular http interceptor handle it.
- * @property {function} displayCall       Optional.  If specified then this function will be called with the fully-formed authorization url when login is invoked.
- * @property {boolean} popUp              Optional; defaults to true.
- * @property {function} callback          Optional.  If specified then this function will be called upon receiving a successful response or error from the authorization server.
- * @property {string} hashPrefix          Optional; defaults to empty string.
- * @property {string} tokenCallbackTimeout Optional; defaults to 6 seconds.  This is the amount of time salte-auth will wait for the identity provider's token endpoint to respond.
+ * Disable certain security validations if your provider doesn't support them.
+ * @typedef {Object} Validation
+ * @property {Boolean} [nonce=true] Passing false will disable nonce validation, leaving you vulnerable to replay attacks.
+ * @property {Boolean} [state=true] Passing false will disable state validation, leaving you vulnerable to XSRF attacks.
+ * @property {Boolean} [azp=true] Passing false will disable azp validation.
+ * @property {Boolean} [aud=true] Passing false will disable aud validation.
  */
 
 /**
- * User information from idtoken.
- *  @class User
- *  @property {string} userName - username assigned from upn, email, or sub (i.e. subject).
- *  @property {object} profile - properties parsed from idtoken.
+ * The configuration for salte auth
+ * @typedef {Object} Config
+ * @property {String} gateway The base url of your identity provider.
```

Comment:
We should probably change this from gateway to identityProviderBaseUrl or something similar.  It is just a happenstance that WSO2 serves as both an identity provider and an API  gateway.